MathLib/InterpolationAlgorithms/PiecewiseLinearInterpolation.cpp

```diff
@@ -14,10 +14,12 @@
 #include <cmath>
 #include <limits>
 #include <utility>
+#include <algorithm>
```

Comment:
please move this before `<cmath>`.  The includes should be sorted by the alphabetical order.